### ISSUE TITLE
Illegal reflective access to field java.lang.ThreadGroup.name

### ISSUE THREAD

**Issue by samuelbernardolip** [nvuillam/vscode-groovy-lint#95]
```
There has been an unexpected error while calling npm-groovy-lint. Please join the end of the logs in Output/GroovyLint if you report the issue
onDidChangeNotification @ breadcrumbsModel.ts:39
```
```
[exthost] [info] ExtensionService#_doActivateExtension NicolasVuillamy.vscode-groovy-lint {"startup":false,"extensionId":{"value":"NicolasVuillamy.vscode-groovy-lint","_lower":"nicolasvuillamy.vscode-groovy-lint"},"activationEvent":"onLanguage:groovy"}
```
```
npm-groovy-lint error: Fatal error while calling CodeNarc
Reason: unknown
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.ReflectionUtils (file:/home/***/.vscode-oss/extensions/nicolasvuillamy.vscode-groovy-lint-1.2.8/server/node_modules/npm-groovy-lint/lib/java/groovy/lib/groovy-3.0.5.jar) to field java.lang.ThreadGroup.name
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.ReflectionUtils
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
java.lang.StringIndexOutOfBoundsException: begin 70, end 72, length 71
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:3319)
	at java.base/java.lang.String.substring(String.java:1874)
	at org.codehaus.groovy.runtime.StringGroovyMethods.getAt(StringGroovyMethods.java:1388)
	at org.codehaus.groovy.runtime.StringGroovyMethods.getAt(StringGroovyMethods.java:1374)
	at org.codehaus.groovy.runtime.dgm$1175.invoke(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:247)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)
	at org.codenarc.util.SourceCodeUtil.sourceLinesBetween(SourceCodeUtil.groovy:60)
	at org.codenarc.util.SourceCodeUtil$sourceLinesBetween.call(Unknown Source)
	at org.codenarc.rule.convention.TrailingCommaAstVisitor.hasTrailingComma(TrailingCommaRule.groovy:90)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:193)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:194)
	at org.codenarc.rule.convention.TrailingCommaAstVisitor.visitMapExpression(TrailingCommaRule.groovy:54)
	at org.codehaus.groovy.ast.expr.MapExpression.visit(MapExpression.java:53)
	at org.codehaus.groovy.ast.GroovyCodeVisitor.lambda$visitListOfExpressions$0(GroovyCodeVisitor.java:202)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at org.codehaus.groovy.ast.GroovyCodeVisitor.visitListOfExpressions(GroovyCodeVisitor.java:202)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitTupleExpression(CodeVisitorSupport.java:249)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitArgumentlistExpression(CodeVisitorSupport.java:367)
	at org.codehaus.groovy.ast.expr.ArgumentListExpression.visit(ArgumentListExpression.java:73)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitMethodCallExpression(CodeVisitorSupport.java:186)
	at org.codehaus.groovy.ast.expr.MethodCallExpression.visit(MethodCallExpression.java:76)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitExpressionStatement(CodeVisitorSupport.java:117)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitExpressionStatement(ClassCodeVisitorSupport.java:200)
	at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:40)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:86)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:164)
	at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:69)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClassCodeContainer(ClassCodeVisitorSupport.java:138)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitConstructorOrMethod(ClassCodeVisitorSupport.java:111)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitMethod(ClassCodeVisitorSupport.java:106)
	at org.codenarc.rule.AbstractAstVisitor.visitMethod(AbstractAstVisitor.java:155)
	at org.codehaus.groovy.ast.ClassNode.visitMethods(ClassNode.java:1100)
	at org.codehaus.groovy.ast.ClassNode.visitContents(ClassNode.java:1093)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClass(ClassCodeVisitorSupport.java:52)
	at org.codenarc.rule.AbstractAstVisitor.visitClass(AbstractAstVisitor.java:132)
	at org.codenarc.rule.AbstractAstVisitorRule.applyTo(AbstractAstVisitorRule.java:97)
	at org.codenarc.rule.AbstractRule.applyTo(AbstractRule.java:143)
	at org.codenarc.rule.Rule$applyTo.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codenarc.rule.Rule$applyTo.call(Unknown Source)
	at org.codenarc.analyzer.AbstractSourceAnalyzer.collectViolations(AbstractSourceAnalyzer.groovy:40)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:193)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:194)
	at org.codenarc.analyzer.FilesystemSourceAnalyzer.processFile(FilesystemSourceAnalyzer.groovy:114)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:351)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:61)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:203)
	at org.codenarc.analyzer.FilesystemSourceAnalyzer$_processDirectory_closure1.doCall(FilesystemSourceAnalyzer.groovy:103)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:263)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1026)
	at groovy.lang.Closure.call(Closure.java:412)
	at groovy.lang.Closure.call(Closure.java:428)
	at org.codehaus.groovy.runtime.ResourceGroovyMethods.eachFile(ResourceGroovyMethods.java:1186)
	at org.codehaus.groovy.runtime.ResourceGroovyMethods.eachFile(ResourceGroovyMethods.java:1204)
	at org.codehaus.groovy.runtime.dgm$1014.invoke(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:247)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)
	at org.codenarc.analyzer.FilesystemSourceAnalyzer.processDirectory(FilesystemSourceAnalyzer.groovy:92)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:193)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:194)
	at org.codenarc.analyzer.FilesystemSourceAnalyzer.analyze(FilesystemSourceAnalyzer.groovy:79)
	at org.codenarc.analyzer.SourceAnalyzer$analyze.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)
	at org.codenarc.CodeNarcRunner.execute(CodeNarcRunner.groovy:74)
	at org.codenarc.CodeNarcRunner$execute.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:130)
	at org.codenarc.CodeNarc.execute(CodeNarc.groovy:171)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:193)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.call(PogoMetaMethodSite.java:73)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)
	at org.codenarc.CodeNarc.main(CodeNarc.groovy:147)

undefined
If you still have an error, post an issue to get help: https://github.com/nvuillam/vscode-groovy-lint/issues
===========================================================================
===========================================================================
2020-12-10T20:33:53.916Z npm-groovy-lint Removed temp file /tmp/npm-groovy-lint/codeNarcTmpDir_0.8813499716169866/Git.groovy as it is not longer used
2020-12-10T20:33:53.918Z npm-groovy-lint Removed temp RuleSet file /tmp/npm-groovy-lint/codeNarcTmpRs_0.5741894723672853.groovy as it is not longer used
2020-12-10T20:33:54.914Z vscode-groovy-lint Request execute command {"command":"groovyLint.lint","arguments":[]}
```

**Comment by nvuillam** [comment]
Would it be possible to have the groovy causing this error ?

**Comment by samuelbernardolip** [comment]
@nvuillam Seems to be the source of the problem...
I'm using archlinux and just updated the OS to be sure that wasn't a problem from any dependency. Then started again and the error persists... Don't know if there was any update to groovy native plugin of vscode.
In the beginning of the error post I have the following config info:
```
"config": {
    "url": "http://localhost:7484/request",
    "method": "post",
    "data": "{\"codeNarcArgs\":\"-basedir=/tmp/npm-groovy-lint/codeNarcTmpDir_0.48361690190812023 -rulesetfiles=file:/tmp/npm-groovy-lint/codeNarcTmpRs_0.6387567318284597.groovy -includes=**/Git.groovy -report=json:stdout\",\"codeNarcBaseDir\":\"/tmp/npm-groovy-lint/codeNarcTmpDir_0.48361690190812023\",\"codeNarcIncludes\":\"**/Git.groovy\",\"parse\":true,\"file\":\"/tmp/npm-groovy-lint/codeNarcTmpDir_0.48361690190812023/Git.groovy\",\"requestKey\":\"/home/***/Git.groovy-none\"}",
(...)
```
I can only find the basedir and all the other files are not present...
What info can I additionaly provide to help with the troubleshoot?

Thanks

**Comment by nvuillam** [comment]
CodeNarc (called by npm-groovy-lint, which is called by VsCode Groovy Lint ) seems to trigger an error
So the issue may be in [CodeNarc](https://github.com/CodeNarc/CodeNarc), but to reproduce i would need your file ( or a short version of it still triggering the same error )

**Comment by samuelbernardolip** [comment]
@nvuillam The project is open source, so you can test it all:
https://github.com/indigo-dc/jenkins-pipeline-library/blob/feature/userRemoteConfigs/src/eu/indigo/scm/Git.groovy
It's the code from branch feature/userRemoteConfigs package eu.indigo.scm.
Thanks
